lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -875,50 +928,66 @@ export class UserAgentApplication {
    */
   // TODO: params to accept AuthRequest object instead
   @resolveTokenOnlyIfOutOfIframe
-  acquireTokenSilent(scopes: Array<string>, authority?: string, user?: User, extraQueryParameters?: string): Promise<string> {
+  acquireTokenSilent(request: AuthenticationParameters): Promise<string> {
     return new Promise<string>((resolve, reject) => {
       // Validate scopes
       // TODO: Replace with new validation pattern
-      const isValidScope = this.validateInputScope(scopes);
+      const isValidScope = this.validateInputScope(request.scopes);
+
       if (isValidScope && !Utils.isEmpty(isValidScope)) {
         // TODO: Reject with custom error here
         // TODO: Is this always accessToken?
         reject(ErrorCodes.inputScopesError + "|" + isValidScope);
         return null;
-      } else {
+      }
+      else {
         // TODO: Remove this from the else block, it is unnecessary
-        if (scopes) {
-          scopes = this.filterScopes(scopes);
+        if (request.scopes) {
+          request.scopes = this.filterScopes(request.scopes);
         }
+        const scope = request.scopes.join(" ").toLowerCase();
+
+        // if the developer passes a userObject give him the priority
+        const userObject = request.account ? request.account : this.getUser();
 
-        const scope = scopes.join(" ").toLowerCase();
-        const userObject = user ? user : this.getUser();
+        // extract if there is an adalIdToken stashed in the cache
         const adalIdToken = this.cacheStorage.getItem(Constants.adalIdToken);
-        //if user is not currently logged in and no login_hint/sid is passed as an extraQueryParamater
-        if (!userObject && Utils.checkSSO(extraQueryParameters) && Utils.isEmpty(adalIdToken) ) {
+
+        //if user is not currently logged in and no login_hint/sid is passed in the request
+        if (!userObject && !!(request.sid  || request.loginHint) && Utils.isEmpty(adalIdToken) ) {
```

Comment:
anotherCheck we can factor out

Reply:
maybe

Reply:
Leaving this as is. Will refactor later if need be.